src/services/refactorProvider.ts

```diff
@@ -14,12 +14,28 @@ namespace ts {
         getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined;
     }
 
-    export interface RefactorContext extends textChanges.TextChangesContext {
+    export interface RefactorOrCodeFixContext {
+        host: LanguageServiceHost;
+        formatContext: ts.formatting.FormatContext;
+    }
+
+    export function getNewLineFromContext(context: RefactorOrCodeFixContext) {
+        const formatSettings = context.formatContext.options;
+        return formatSettings ? formatSettings.newLineCharacter : context.host.getNewLine();

```

Comment:
The `newLineCharacter` property is declared optional, so this should return `context.host.getNewLine()` if that's undefined.
Although it looks like `getNewLine()` is also optional -- see `getNewLineOrDefaultFromHost`.

Reply:
Updated